### PR TITLE
unit test added

### PR DESCRIPTION
**Pull request by nickyanalytics**
This PR introduce a new event205 "AS Internal". It's called if a page was loaded from another AS Brand (except used Outbrain Widgets). This event is needed for the new Tealium Marketing Channel setup.

### CODE REVIEW

File: extensions/doPlugins_global.js [hunk 1]
```diff
@@ -185,6 +185,18 @@ s._articleViewTypeObj = {
         }
     },
 
+    isFromAsDomain: function (referrer) {
+        //for bild domains see isFromBild & isFromBildMobile
+        const asDomains = ['welt.de','fitbook.de','stylebook.de','techbook.de','travelbook.de','myhomebook.de','bz-berlin.de','rollingstone.de','metal-hammer.de','musikexpress.de','petbook.de'];
+        const referringDomain = s._utils.getDomainFromURLString(referrer);
+        const isNotInternal = this.isFromInternal(referrer);
```
Comment (mehakraza) [inline extensions/doPlugins_global.js:192]: As the const is set to the result of isFromInternal, I think the name 'isInternal' would better represent what it is.

File: tests/doplugins/doplugins_article_view_type.test.js [hunk 2]
```diff
@@ -179,7 +179,31 @@ describe('articleViewType()', () => {
...
+        const asDomains = 'fitbook.de';
+        let getDomainFromURLStringMock;
+    
+        beforeEach(() => {
+            getDomainFromURLStringMock = jest.spyOn(s._utils, 'getDomainFromURLString').mockReturnValue('');
+        });
+    
+        it('should return TRUE if referrer is from internal Domain (except bild.de, like from fitbook.de)', function () {
+            getDomainFromURLStringMock.mockReturnValue(asDomains);
+            const result = s._articleViewTypeObj.isFromAsDomain(anyReferrer);
```
Comment (mehakraza) [inline tests/doplugins/doplugins_article_view_type.test.js:194]: May I ask why isFromAsDomain is called here with 'anyReferrer', if we're checking for the truthy case should we call it with 'asDomains'?

File: tests/doplugins/doplugins_article_view_type.test.js [hunk 3]
```diff
@@ -179,7 +179,31 @@ describe('articleViewType()', () => {
...
+            expect(getDomainFromURLStringMock).toHaveBeenLastCalledWith(anyReferrer);
+            expect(result).toBe(true);
+        });
+    
+        it('should return FALSE if referrer is NOT from internal Domain (except bild.de, like from fitbook.de)', function () {
+            getDomainFromURLStringMock.mockReturnValue('any-other-domain.com');
```
[the remark this answers is not in the document]
Comment (nickyanalytics) [reply tests/doplugins/doplugins_article_view_type.test.js:200]: Copy & paste.... line is deleted

File: tests/doplugins/doplugins_article_view_type.test.js (hunk 2 above)
Comment (nickyanalytics) [reply tests/doplugins/doplugins_article_view_type.test.js:194]: Oh yes, today I see it :-). Thank you

**Review by mehakraza — APPROVED** [review]
Looks good 👍

**Pull request merged**